### DISCUSSION TITLE
Error "handshake: Can't find fingerprint". What does it mean? What is missing?

### DISCUSSION THREAD

**Discussion by Kalinin-Andrey** [xelaj/mtproto#131] [Q&A]
Hello, everyone!
I`m trying to create a client
```
client, err := telegram.NewClient(telegram.ClientConfig{ ...
```
in file telegram/common.go
on string `err = m.CreateConnection()`
I have an error "handshake: Can't find fingerprint".
What does it mean? What is missing?

Debug IDE in place with error (handshake.go):
![image](https://user-images.githubusercontent.com/8151806/178973433-0a91394f-06ad-4cca-84aa-450d1bee1c9c.png)

As you can see variable `m.publicKey` is set and `res.Fingerprints` has been calculated.

Why are the calculation results not equal? And why `b` is negative?
![image](https://user-images.githubusercontent.com/8151806/178976876-ad6d3397-7875-495b-ac88-b9d4e2fbd522.png)

**Comment by TinGimranov** [comment]
Hello! Have you found out what’s the problem?
